### PR TITLE
[wip] introduce filter helpers

### PR DESCRIPTION
**Pull request by RonnyPfannschmidt**

### CODE REVIEW

**Comment by mfalesni** [comment]
So, maybe more comments to introduce what exactly is this PR supposed to do would be awesome. Especially with an example in the comment. If this corresponds to some issue, it should be linked with `Closes #issuenumber`.

File: src/deprecate/filters.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    warnings.warn('the filter registration api is subject to change',
+                  category=FutureWarning,
+                  stacklevel=2, )
+
+    def filter_registring_decorator(category):
+        assert issubclass(category, Warning)
+        _module.simplefilter(action, category)
+        return category
+
+    return filter_registring_decorator
```
Comment (mfalesni) [inline src/deprecate/filters.py:27]: functools.wraps?

Comment (RonnyPfannschmidt) [inline src/deprecate/filters.py:27]: not needed, this is a pass-trough

Comment (mfalesni) [inline src/deprecate/filters.py:27]: right

**Comment by RonnyPfannschmidt** [comment]
closing as im no longer working on it

**Pull request closed without merging**